core/tests/mem_core_tests/src/concurrent_mdlw_event_emitter_test.cpp

```diff
@@ -311,15 +342,22 @@ TEST(ConcurrentEventEmitterTest, WithObjectPoolWithFailuresNonBlocking) {
         while (!(provider._internalQueue.size_approx()==0)) {
             std::this_thread::sleep_for(std::chrono::milliseconds(1));
         }
-        provider.stop();
     });
 
     t2.join();
 
+    int timeout = 0;
+    while (timeout < 100 && eventListener.counter < queueSize)
```

Comment:
I don't think this line is needed if we have ensured (on line 342 of this new file) that the internalQueue is empty. As per the implementation, publisher fills the internalQueue and the polling loop empties it and runs the different listeners. So if the internalQueue is empty, then all published events have been accounted for.

Has this particular test failed in the builds? I think the code block on line 342 would be sufficient rather than having a sleep block with an arbitrarily chosen timeout.

Reply:
This test was failing 50-50 in Jenkins. This is the reason for the fix.
@guillermo-sarabia : ```timeout``` as variable name is misleading. We should call it ```attemptCounter``` or even better we convert ```timeout``` into an actual time measuring variable using ```std::chrono``` (https://en.cppreference.com/w/cpp/chrono) and define the max sleep time to be 0.5 secs

Reply:
@mharshe I'm onboard with the checking the internal queue instead of  `eventListener.counter < queueSize`. However, a timeout / attempt counter is still neccesary, as errors might cause the test to hang (and occupy) jenkins. 
@pablogh-2000 I will modify it.